src/components/CreateGraph/CreateGraph.jsx

```diff
@@ -30,131 +27,181 @@ const CreateGraph = () => {
-              icon={button.icon}
-              onClick={button.onClick}
+    const addEdge = (sourceNode, targetNode) => {
+
+        const sourceNodeId = sourceNode.key;
+        const targetNodeId = targetNode.key;
+
+        const edgeExists = edges.some(
+            (edge) =>
+                (edge.source.key === sourceNodeId && edge.target.key === targetNodeId) ||
+                (edge.source.key === targetNodeId && edge.target.key === sourceNodeId)
+        );
+
+        if (edgeExists) {
+            message.error(texts.EDGE_EXISTS);
+            return;
+        }
+        if (sourceNodeId === targetNodeId) {
+            return;
+        }   
+
+        const newEdge = { source: sourceNode, target: targetNode, isSelected: false, key: edges.length + 1 };
+        setEdges([...edges, newEdge]);
+        setSelectedEdge(newEdge);
+    };
+
+
+    const deleteNode = () => {
+        if (!selectedNode) {
+            message.error(texts.NO_NODE_SELECTED);
+            return;
+        }
+
+        const updatedNodes = nodes.filter((node) => !(node.x === selectedNode.x && node.y === selectedNode.y));
+        setNodes(updatedNodes);
+        setSelectedNode(null);
+    };
+
+    const deleteEdge = () => {
+        if (!selectedEdge) {
+            message.error(texts.NO_EDGE_SELECTED);
+            return;
+        }
+
+        const updatedEdges = edges.filter((edge) => {
+            return !(edge.source === selectedEdge.source && edge.target === selectedEdge.target) &&
+                !(edge.source === selectedEdge.target && edge.target === selectedEdge.source);
+        });
+
+        setEdges(updatedEdges);
+        setSelectedEdge(null);
+    };
+
+    const clearAll = () => {
+        setNodes([]);
+        setEdges([]);
+        setSelectedEdge(null);
+        setSelectedNode(null);
+    };
+
+    const buttonsData = [
+        { text: 'Add Node', icon: <PlusCircleFilled />, onClick: handleStartPlacingNode },
+        { text: 'Delete Node', icon: <MinusCircleFilled />, onClick: deleteNode },
+        { text: 'Delete Edge', icon: <MinusCircleFilled />, onClick: deleteEdge },
+        { text: 'Clear All', icon: <DeleteFilled />, onClick: clearAll }
+    ];
+
+    return (
+        <>
+            <Flex gap="L" align="center" vertical>
+                <Flex gap="small" wrap="wrap">
+                    {buttonsData.map((button, index) => (
+                        <Button key={index} type="primary" size="large" icon={button.icon} onClick={button.onClick}>
+                            {button.text}
+                        </Button>
+                    ))}
+                </Flex>
+            </Flex>
+
+            <div
+                onClick={handleNodePlacement}
+                style={{
+                    position: 'relative',
+                    width: '500px',
+                    height: '500px',
+                    border: '1px solid black',
+                    margin: '20px auto'
+                }}
```

Comment:
Why are you using an inline style for a simple div? 

If it not ant usage please use CSS files